### PR TITLE
[Merged by Bors] - feat: gcongr works with eta and proofs

### PR DESCRIPTION
**Pull request by fpvandoorn**
* `gcongr` now recognizes eta-expanded variables as variables
* `gcongr` now recognizes all proofs as "defeq" even if the proofs have different types (in e.g. `Finset.sup'_mono`
* Give a bit more information in the error message when not accepting a `gcongr`-lemma.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/GCongr/inequalities.lean [hunk 1]
```diff
@@ -169,6 +170,12 @@ example {a b c d e : ℝ} (_h1 : 0 ≤ b) (_h2 : 0 ≤ c) (hac : a * b + 1 ≤ c
   guard_target =ₛ a * b ≤ c * d
   linarith
 
+-- test big operators
+example (f g : ℕ → ℕ) (s : Finset ℕ) (h : ∀ i ∈ s, f i ≤ g i) :
```
Comment (grunweg) [inline test/GCongr/inequalities.lean:174]: Do f, g and s need to be explicit? (I am aware the same question applies to the next example also.)

**Review by grunweg — COMMENTED** [review]
It's great to see tactic improvements, thanks for doing this!
I have one tiny question about a test. And can you add a descriptive PR title, please? Thanks!

File: test/GCongr/inequalities.lean (hunk 1 above)
Comment (fpvandoorn) [reply test/GCongr/inequalities.lean:174]: No, that is irrelevant

Comment (grunweg) [reply test/GCongr/inequalities.lean:174]: :+1: In this case: do you mind switching them to implicit? (If this were a mathlib file, this is what I'd ask a PR author to do.)

Comment (j-loreaux) [reply test/GCongr/inequalities.lean:174]: I don't think this really matters for test files. This is not something we're careful about at all (see for instance, the rest of the this file).

**Comment by j-loreaux** [comment]
This looks simple enough for even me to understand.

maintainer merge

File: test/GCongr/inequalities.lean (hunk 1 above)
Comment (grunweg) [reply test/GCongr/inequalities.lean:174]: Sure, I'm happy to adjust to that policy as well.

**Comment by dupuisf** [comment]
Looks good!

bors r+

**Pull request closed without merging**